apps/finance/app/src/components/Transfers.js

```diff
@@ -12,252 +13,248 @@ import {
+        entity,
+        token,
+      }) => {
+        const { symbol, decimals } = tokenDetails[toChecksumAddress(token)]
+        const formattedAmount = formatTokenAmount(
+          amount,
+          isIncoming,
+          decimals,
+          true,
+          { rounding: 5 }
         )
-      )
-      .join('\n')
-    return window.encodeURI(csvContent)
-  }
-  getCsvFilename = () => {
-    const { selectedDateRange } = this.state
-    const start = format(selectedDateRange.start, 'yyyy-MM-dd')
-    const end = format(selectedDateRange.end, 'yyyy-MM-dd')
-    return `transfers_${start}_to_${end}.csv`
-  }
-  showMoreTransfers = () => {
-    this.setState(prevState => ({
-      displayedTransfers: prevState.displayedTransfers + TRANSFERS_PER_PAGE,
-    }))
+        const { name = '' } = (await resolveAddress(entity)) || {}
+        return `${formatDate(
+          date
+        )},${name},${entity},${reference},${`${formattedAmount} ${symbol}`}`
+      }
+    )
+  )
+  const csvContent = [
+    'data:text/csv;charset=utf-8,Date,Name,Source/Recipient,Reference,Amount',
+  ]
+    .concat(mappedData)
+    .join('\n')
+  return window.encodeURI(csvContent)
+}
+const getDownloadFilename = (dao, { start, end }) => {
+  const today = format(Date.now(), 'yyyy-MM-dd')
+  let filename = `finance_${dao}_${today}.csv`
+  if (start && end) {
+    const formattedStart = format(selectedDateRange.start, 'yyyy-MM-dd')
+    const formattedEnd = format(selectedDateRange.end, 'yyyy-MM-dd')
+    filename = `finance_${dao}_${formattedStart}_to_${formattedEnd}.csv`
   }
+  return filename
+}
 
-  // Filter transfer based on the selected filters
-  getFilteredTransfers({
-    tokens,
+const Transfers = React.memo(({ dao, tokens, transactions }) => {
+  const { below } = useViewport()
+  const compactMode = below('medium')
+  const [filtersOpened, setFiltersOpened] = React.useState(!compactMode)
+  const [selectedToken, setSelectedToken] = React.useState(0)
+  const [displayedTransfers, setDisplayedTransfers] = React.useState(
+    INITIAL_TRANSFERS_PER_PAGE
+  )
+  const [selectedTransferType, setSelectedTransferType] = React.useState(0)
+  const [selectedDateRange, setSelectedDateRange] = React.useState({
+    start: null,
+    end: null,
+  })
+  const handleToggleFiltersClick = React.useCallback(() => {
+    setFiltersOpened(!filtersOpened)
+  })
+  const handleTokenChange = React.useCallback(index => {
+    setSelectedToken(index)
+    setDisplayedTransfers(INITIAL_TRANSFERS_PER_PAGE)
+  })
+  const handleTransferTypeChange = React.useCallback(index => {
+    setSelectedTransferType(index)
+    setDisplayedTransfers(INITIAL_TRANSFERS_PER_PAGE)
+  })
+  const handleResetFilters = React.useCallback(() => {
+    setDisplayedTransfers(INITIAL_TRANSFERS_PER_PAGE)
+    setSelectedToken(0)
+    setSelectedTransferType(0)
+  })
+  const showMoreTransfers = React.useCallback(() => {
+    setDisplayedTransfers(displayedTransfers + INITIAL_TRANSFERS_PER_PAGE)
+  })
```

Comment:
This is probably starting to get annoying... but I just noticed these callbacks also need to have their dependencies listed :smile: